lib/cli.js

```diff
@@ -182,16 +182,11 @@ exports.run = () => {
 	if (conf.watch) {
 		try {
 			const watcher = new Watcher(logger, api, files, arrify(conf.source));
-			watcher.observeStdin(process.stdin);
+			watcher.observeStdin(process.stdin).then(() => {
+				logger.exit(0);
```

Comment:
Instead of exiting directly, can we stop all workers and let the process exit on its own?

Reply:
Sure. How to stop those workers?
By the way, I saw in `test-worker.js`:
```js
process.on('ava-exit', () => { process.exit(0); })
```
So the watcher should fire that even in all sub-processes?

Reply:
Not quite. The child processes should be [sent the `SIGINT` signal](https://nodejs.org/api/child_process.html#child_process_subprocess_kill_signal). See https://github.com/avajs/ava/blob/bb91862eaac1a3e89f651afb15ec9691745ee222/lib/fork.js.